docs/developers/testing.rst

```diff
@@ -0,0 +1,110 @@
+==================
+Run MicroOVN tests
+==================
+
+MicroOVN has two types of tests, linter checks and functional tests and this
+page will show how to run them.
+
+Linter checks
+-------------
+
+Linting is currently applied only to the tests themselves, not the main
+codebase. The prerequisites for running linter are:
+
+* ``make``
+* ``shellcheck``
+
+You can install them with:
+
+.. code-block:: none
+
+   sudo apt install make shellcheck
+
+To perform linting, go into the repository's root directory and run:
+
+.. code-block:: none
+
+   make check-lint
+
+Functional tests
+----------------
+
+These tests build the MicroOVN snap and use it to deploy the ``OVN`` cluster
+in LXD containers. This cluster is then used for running functional test
+suites.
+
+Functional test requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There is no need to run tests in dedicated VMs or in isolated environments as
+all functional tests run inside containers and no changes are made to the host
+running them.
+
+Functional tests need to build the MicroOVN snap and run multiple containers
+via LXD, so make sure you first complete steps and requirements for
+:doc:`building MicroOVN from source <building>`.
+
+The final step is to ensure that you have installed required
```

Comment:
Secondly, ensure that you have installed `Bash Automated... (BATS)`, a software dependency. Due...